ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1026,7 +1026,7 @@ private struct ZipState<Left, Right> {
 	var completed: (left: Bool, right: Bool) = (false, false)
 
 	var isFinished: Bool {
-		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty)
+		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty) || (completed.left && completed.right)
```

Comment:
Could you open up a PR against `master` for this? It's unrelated to the other changes here, but more importantly it's not specific to `RAC5`

Also: If `completed.left && completed.right` and the previous conditions aren't true, that means that there are more values that should have been sent out. So the fix should be to send those values, not to prematurely end the signal.

This must be fallout from #2917. Major bonus point if you can add a test. 😄